### ISSUE TITLE
Path parameters not encoded

### ISSUE THREAD

**Issue by rchl** [ferdikoomen/openapi-typescript-codegen#938]
**Describe the bug**
For endpoints that take the parameter and use it within the API path, the parameters needs to be encoded using `encodeURIComponent`.

Currently that's not the case so for example if the endpoint's path is
```js
`/aaaa/${userId}/ccccc`
```
and someone passes `bbbbb/xxxxx` as a `userId`, it will result in calling a different or non-existent endpoint. Possibly even creating security issues...

**Comment by ferdikoomen** [comment]
@rchl the `encodeURIComponent` is actually too strict in some cases. For instance when looking at Wikipedia's URL system: https://commons.wikimedia.org/wiki/Special:RecentChanges you will see a `:` character in the path. The `encodeURIComponent` method converts this too `%3A`, that is not what we want. Although some webservers will convert this back to`:`, this is not always the case. I will keep the ticket open to see if there is a good solution for this.

**Comment by rchl** [comment]
I suppose it might make sense to mimic the behavior of expressjs and its handling of route parameters https://expressjs.com/en/guide/routing.html#route-parameters

It's likely robust and world tested enough to be safe.

**Comment by rchl** [comment]
Well, expressjs doens't deal with encoding, only encoding so maybe not relevant.

**Comment by ferdikoomen** [comment]
@rchl I just pushed a new version 0.16.0 that will do a `encodeURI` on properties in the path, that will not solve your `bbbbb/xxxxx` as a `userId` issue. But you can make the encoder more strict by setting:
```
OpenAPI.ENCODE = encodeURIComponent; // or anything custom that accepts a string and returns a string ;-)
```
Then it will use that encode method for the properties in the path, so `bbbbb/xxxxx` becomes `bbbbb%2Fxxxxx`

**Comment by rchl** [comment]
Thanks a lot.

Not that important but I don't see this being mentioned anywhere in the readme or changelog. It's hard to follow the project changes when trying to upgrade to latest versions :)
Also the fact that I have custom request.ts makes it hard to upgrade due to many changes in the default request implementation.

That all means it will take some time before I get to testing it. :)

**Comment by ferdikoomen** [comment]
@rchl Why are you modifying the default request implementation? What do you need? Maybe i can help you with that? Iw ill add a section to the documentation about the different props in the OpenAPI object aswell.

**Comment by rchl** [comment]
The initial reason was just to use `axios` for requests rather than `fetch` but it seems like generating `axios`-based request handler is also supported now. I will have to look into using it then instead.
